src/main/java/com/ksm/wstbackoffice/controller/CountryController.java

```diff
@@ -44,4 +47,20 @@ public CountryController(CountryService countryService) {
         return ResponseEntity.status(HttpStatus.OK)
                 .body(country);
     }
+
+    @PostMapping("/start-with")
+    public ResponseEntity<Map<String, HashMap<String, ?>>> findCountriesStartWith(@RequestBody List<String> stringsToStartWith) {
+        if (stringsToStartWith.isEmpty()) {
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(null);
+        }
+
+        Map<String, HashMap<String, ?>> filteredCountries = countryService.startWith(stringsToStartWith);
+
+        if (filteredCountries.isEmpty()) {
+            return ResponseEntity.status(HttpStatus.NOT_FOUND).body(null);
```

Comment:
So, the general rule is
You return NOT_FOUND if the user is getting/updating/deleting/etc something by id.
When a user is filtering results or expecting to receive many values (not one by id) you should return an empty value to him.
That is why findAll() never give NOT_FOUND but returns emptyList() with OK status code.
Something similar we should do without giving NOT_FOUND
Saying that can you find a better solution here?

Reply:
done